state/request_options.go

```diff
@@ -0,0 +1,49 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package state
+
+import (
+	"fmt"
+)
+
+const (
+	FirstWrite = "first-write"
+	LastWrite  = "last-write"
+	Strong     = "strong"
+	Eventual   = "eventual"
+)
+
+// CheckSetRequestOptions checks if set request options use supported keywords
+func CheckSetRequestOptions(req *SetRequest) error {
+	if req.Options.Concurrency != "" && req.Options.Concurrency != FirstWrite && req.Options.Concurrency != LastWrite {
+		return fmt.Errorf("unrecognized concurrency model '%s'", req.Options.Concurrency)
+	}
+	if req.Options.Consistency != "" && req.Options.Consistency != Strong && req.Options.Consistency != Eventual {
+		return fmt.Errorf("unrecognized consistency model '%s'", req.Options.Consistency)
+	}
+	return nil
+}
+
+// CheckDeleteRequestOptions checks if delete request options use supported keywords
+func CheckDeleteRequestOptions(req *DeleteRequest) error {
```

Comment:
This is a copy paste of `CheckSetRequestOptions`, can we merge?

Reply:
@yaron2 , these 2 methods are not touched in this changelist as I removed retries only, but I agree these should be changed/merged for sure.
For that, we need to introduce common interface for request. I can introduce new interface and make a change in separate PR. 
Should then I consider that change in different PR ? Please let me know.

Reply:
I would tackle this in this PR.

Reply:
done.